TFT/src/User/API/UI/GUI.c

```diff
@@ -605,84 +606,58 @@ void GUI_DispOne(int16_t sx, int16_t sy, const CHAR_INFO *pInfo)
 {
   if (pInfo->bytes == 0) return;
 
-  uint8_t x = 0,
-          y = 0,
-          j = 0,
-          i = 0;
+  uint8_t x = 0;
+  uint8_t y = 0;
+  uint8_t j = 0;
+  uint8_t i = 0;
   uint16_t bitMapSize = (pInfo->pixelHeight * pInfo->pixelWidth / 8);
   uint8_t  font[bitMapSize];
   uint32_t temp = 0;
 
   W25Qxx_ReadBuffer(font, pInfo->bitMapAddr, bitMapSize);
 
-  if (guiTextMode == GUI_TEXTMODE_TRANS)
+  for (x = 0; x < pInfo->pixelWidth; x++)
```

Comment:
this function was split just to improve performance (avoiding, for every pixel, two extra "if then else" to apply transparency, fixed bk color or bk image). It was only a bunch of bytes more but better performance. In a slower TFT HW (e.g. TFT28) the stuttering on status screen (e.g a scrolling text in the info box) is evident (at least two frames were out of synch)

Reply:
That's why I have uses the switch statement, which is just like the `goto` statement instead of comparing every possibility like if-else. There is no change in performance with this

Reply:
ok good. It was an explanation of my choice (a PR older than 1 year)